vcfconcat.c

```diff
@@ -549,6 +553,114 @@ static void concat(args_t *args)
     }
 }
 
+BGZF *hts_get_bgzfp(htsFile *fp);
+static void naive_concat(args_t *args)
+{
+    // only compressed BCF atm
+
+    BGZF *bgzf_out;
+    if ( strcmp("-",args->output_fname) )
+        bgzf_out = bgzf_open(args->output_fname,"w");
+    else
+        bgzf_out = bgzf_dopen(fileno(stdout), "w");
+
+    int page_size = getpagesize();
+    char *buf = (char*) valloc(page_size);
+    kstring_t tmp = {0,0,0};
+    int i;
+    for (i=0; i<args->nfnames; i++)
+    {
+        htsFile *hts_fp = hts_open(args->fnames[i],"r");
+        if ( !hts_fp ) error("Failed to open: %s\n", args->fnames[i]);
+        htsFormat type = *hts_get_format(hts_fp);
+
+        if ( type.format==vcf ) error("The --naive option currently works only for compressed BCFs, sorry :-/\n");
+        if ( type.compression!=bgzf ) error("The --naive option currently works only for compressed BCFs, sorry :-/\n");
+
+        BGZF *fp = hts_get_bgzfp(hts_fp);
+        if ( !fp || bgzf_read_block(fp) != 0 || !fp->block_length )
+            error("Failed to read %s: %s\n", args->fnames[i], strerror(errno));
+
+        uint8_t magic[5];
+        if ( bgzf_read(fp, magic, 5)<0 ) error("Failed to read the BCF header in %s\n", args->fnames[i]);
+        if (strncmp((char*)magic, "BCF\2\2", 5) != 0) error("Invalid BCF magic string in %s\n", args->fnames[i]);
+
+        bgzf_read(fp, &tmp.l, 4);
+        hts_expand(char,tmp.l,tmp.m,tmp.s);
+        bgzf_read(fp, tmp.s, tmp.l);
+
+        // write only the first header
+        if ( i==0 )
+        {
+            if ( bgzf_write(bgzf_out, "BCF\2\2", 5) !=5 ) error("Failed to write %d bytes to %s\n", 5,args->output_fname);
+            if ( bgzf_write(bgzf_out, &tmp.l, 4) !=4 ) error("Failed to write %d bytes to %s\n", 4,args->output_fname);
+            if ( bgzf_write(bgzf_out, tmp.s, tmp.l) != tmp.l) error("Failed to write %d bytes to %s\n", tmp.l,args->output_fname);
+        }
+
+        // Output all non-header data that were read together with the header block
+        int nskip = fp->block_offset;
+        if ( fp->block_length - nskip > 0 )
+        {
+            if ( bgzf_write(bgzf_out, fp->uncompressed_block+nskip, fp->block_length-nskip)<0 ) error("Error: %d\n",fp->errcode);
+        }
+        if ( bgzf_flush(bgzf_out)<0 ) error("Error: %d\n",bgzf_out->errcode);
+
+
+        // Stream the rest of the file as it is, without recompressing, but remove BGZF EOF blocks
+        ssize_t nread, ncached = 0, nwr;
+        const int neof = 28;
+        char cached[neof];
```

Comment:
No need to use a C99 VLA here.

Really it would be better to use `bgzf_check_EOF` or the like rather than needing to know what a trailer block looks like here.